src/Components/IndexScene/ShowLogsButtonScene.tsx

```diff
@@ -37,13 +37,15 @@ export class ShowLogsButtonScene extends SceneObjectBase<ShowLogsButtonSceneStat
     });
   }
 
-  onClick = () => {
+  getLink = () => {
     const labelsVar = getLabelsVariable(this);
     const positiveFilter = labelsVar.state.filters.find((f) => isOperatorInclusive(f.operator));
 
     if (positiveFilter) {
-      navigateToInitialPageAfterServiceSelection(positiveFilter.key, positiveFilter.value);
+      return getDrillDownIndexLink(positiveFilter.key, positiveFilter.value);
     }
+
+    return '';
   };
 
```

Comment:
This feels wrong. I would consider something like:

```
const link = model.getLink();
if (!link) {
  return null;
}
return <LinkButton ... />
```

Reply:
Take a look at https://github.com/grafana/logs-drilldown/pull/1106/commits/5c9a2818b281b7bb15203c8ce660d85082a8af7f